packages/next/src/app-dir/server.ts

```diff
@@ -26,31 +28,46 @@ import { cache } from 'react';
 import { formDataToObject } from './formDataToObject';
 import type {
   ActionHandlerDef,
-  CreateTRPCNextAppRouterOptions,
+  CreateTRPCNextAppRouterServerOptions,
   inferActionDef,
 } from './shared';
 import { generateCacheTag, isFormData } from './shared';
-import type { NextAppDirDecorateRouterRecord } from './types';
+import type {
+  NextAppDirDecorateRouterRecord,
+  NextAppDirRuntime,
+} from './types';
 
 // ts-prune-ignore-next
 export function experimental_createTRPCNextAppDirServer<
   TRouter extends AnyRouter,
->(opts: CreateTRPCNextAppRouterOptions<TRouter>) {
-  const getClient = cache(() => {
+>(opts: CreateTRPCNextAppRouterServerOptions<TRouter>) {
+  const getClient = cache(async (callOpts: ProxyCallbackOptions) => {
     const config = opts.config();
-    return createTRPCUntypedClient(config);
+    const client = createTRPCUntypedClient(config);
+    const runtime = client.runtime as NextAppDirRuntime<TRouter>;
+
+    const ctx = await config.createContext();
+    runtime.ctx = ctx;
+    runtime.cacheTagSeparators = config.contextSelector?.(ctx, callOpts) ?? [];
```

Comment:
I dont think we can cache this anymore when it's procedure-specific, it should be invoked every call i think